transforms/helpers/options.ts

```diff
@@ -1,49 +1,98 @@
+import { inspect } from 'node:util';
 import type { ZodError } from 'zod';
 import { z } from 'zod';
 import logger from './log-helper';
 import type { RuntimeData } from './runtime-data';
+import {
+  StringArraySchema,
+  StringBooleanSchema,
+  StringFalseSchema,
+} from './schema-helper';
 
 const DEFAULT_DECORATOR_CONFIG = {
   inObjectLiterals: [],
 };
 
 const DecoratorOptionsSchema = z
   .object({
-    /** Allow-list for decorators currently applied to object literal properties that can be safely applied to class properties. */
-    inObjectLiterals: z.preprocess((arg: unknown) => {
-      return typeof arg === 'string' ? arg.split(/\s*,\s*/) : arg;
-    }, z.array(z.string())),
+    inObjectLiterals: StringArraySchema.describe(
+      'Allow-list for decorators currently applied to object literal properties that can be safely applied to class properties.'
```

Comment:
I find this message pretty confusing. What exactly is it trying to say?

Reply:
Input:
https://github.com/ember-codemods/ember-native-class-codemod/blob/f8d8fd237dcb4fbbb213d15e5ca09bc1a681d992/transforms/ember-object/__testfixtures__/options/object-literal-decorators.input.js#L3-L6

Config:
https://github.com/ember-codemods/ember-native-class-codemod/blob/f8d8fd237dcb4fbbb213d15e5ca09bc1a681d992/transforms/ember-object/__testfixtures__/options/object-literal-decorators.options.json#L1-L5

Output:
https://github.com/ember-codemods/ember-native-class-codemod/blob/f8d8fd237dcb4fbbb213d15e5ca09bc1a681d992/transforms/ember-object/__testfixtures__/options/object-literal-decorators.output.js#L4-L8

Without the config, the `@userAdded` decorator would not transform at all because it's applied to a property and not a method.

Reply:
What about something like:

```text
A list of decorators that are allowed in object literals. When the codemod finds a field with one of these decorators, it will be translated directly into a class field with the same decorator. Including a decorator in this list means that you believe that the decorator will work when used in a class.
```